lib/maths/CBoostedTreeImpl.cc

```diff
@@ -1530,7 +1534,8 @@ void CBoostedTreeImpl::recordHyperparameters() {
     m_Instrumentation->hyperparameters().s_ClassAssignmentObjective = m_ClassAssignmentObjective;
     m_Instrumentation->hyperparameters().s_DownsampleFactor = m_DownsampleFactor;
     m_Instrumentation->hyperparameters().s_NumFolds = m_NumberFolds;
-    m_Instrumentation->hyperparameters().s_TrainFractionPerFold = m_TrainFractionPerFold;
+    m_Instrumentation->hyperparameters().s_NumTrainingRows =
```

Comment:
NumTrainingRows is, strictly speaking, not a hyperparameter. I am fine with this for now, but maybe you can add a TODO to move it into the `metadata` part of `m_Instrumentation` i the future.

Reply:
This is a good suggestion. I just went ahead and moved it rather than leaving a TODO. See [e0a61bf](https://github.com/elastic/ml-cpp/pull/1941/commits/e0a61bfe43abdac2c4a82e97e29301c2a412398c).